tensorflow_datasets/core/_sharded_files.py

```diff
@@ -26,9 +26,15 @@
 from __future__ import division
 from __future__ import print_function
 
+from typing import Dict, List, Union
 
-def get_read_instructions(from_, to, filenames, shard_lengths,
-                          shardref_name="filename"):
+
+def get_read_instructions(
+    from_: int,
+    to: int,
+    filenames: List[str],
+    shard_lengths: List[int],
+    shardref_name: str = "filename") -> List[Dict[str, Union[str, int]]]:
```

Comment:
I think this could be aliased with ` -> List[InstructionDict]` and:

```python
# TODO(b/134526091): Should use TypedDict once Python 3.8 supported.
InstructionDict = Dict[str, Union[str, int]]
```
I think this is more readable and will simplify switching to TypedDict: https://www.python.org/dev/peps/pep-0589/

Reply:
Done